src/buffer/out/TextAttribute.cpp

```diff
@@ -210,11 +246,51 @@ void TextAttribute::Debolden() noexcept
     _SetBoldness(false);
 }
 
+void TextAttribute::EnableUnderline() noexcept
+{
+    SetBottomHorizontalDisplayed(true);
+}
+
+void TextAttribute::DisableUnderline() noexcept
+{
+    SetBottomHorizontalDisplayed(false);
+}
+

```

Comment:
Is there any reason why these two couldn't be a single `SetUnderline` method taking a bool, the same as most of the others (`SetItalics`, `SetBlinking`, etc)? Having them as two separate methods just seems to make the calling code unnecessarily complicated (ideally the code below would be a single call).

https://github.com/microsoft/terminal/blob/84b1b874dd09dd11c5099169acfcbb28c7ce0609/src/types/sgrStack.cpp#L156-L163

I'd also love it if we could get rid of `Embolden` and `Debolden`, and add a proper setter for the `Invert`, for the same reason, although I don't know if that would be considered out of scope for this PR.

Reply:
Okay, I went ahead and tackled bold and underline for this latest revision.

Reply:
If we add support for double size fonts, can have Embiggen and Debiggen methods? can we can we can we